zcash_primitives/src/note_encryption.rs

```diff
@@ -1019,9 +1023,9 @@ mod tests {
             canopy_activation_height,
             canopy_activation_height + ZIP212_GRACE_PERIOD,
         ];
-        let leadbyte_array = [0x02, 0x03, 0x01];
+        let leadbytes = [0x02, 0x03, 0x01];
 
-        for (i, &height) in heights.iter().enumerate() {
+        for (_i, (&height, &leadbyte)) in heights.iter().zip(leadbytes.iter()).enumerate() {
```

Comment:
If the index is now unused, we can just remove it (here and below):
```suggestion
        for (&height, &leadbyte) in heights.iter().zip(leadbytes.iter()) {
```